scripts/quests/bastok/Chips.lua

```diff
@@ -0,0 +1,136 @@
+                    elseif
+                        player:hasCompletedMission(xi.mission.log_id.COP, xi.mission.id.cop.ONE_TO_BE_FEARED) and
+                        not quest:getMustZone(player)
+                    then
+                        return quest:progressEvent(169)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [169] = function(player, csid, option, npc)
+                    if option == 0 then
+                        quest:begin(player)
+                    elseif option == 1 then
+                        quest:setMustZone(player)
+                    end
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_ACCEPTED
+        end,
+
+        [xi.zone.METALWORKS] =
+        {
+            ['Cid'] =
+            {
+                onTrade = function(player, npc, trade)
+                    if
+                        player:hasItem(xi.item.CCB_POLYMER) and
+                        npcUtil.tradeHasExactly(trade, { xi.item.CARMINE_CHIP, xi.item.CYAN_CHIP, xi.item.GRAY_CHIP })
+                    then
+                        player:messageSpecial(metalID.text.ITEM_CANNOT_BE_OBTAINED, xi.item.CCB_POLYMER)
+                    elseif
+                        player:getFreeSlotsCount() == 0 and
+                        npcUtil.tradeHasExactly(trade, { xi.item.CARMINE_CHIP, xi.item.CYAN_CHIP, xi.item.GRAY_CHIP })
+                    then
+                        player:messageSpecial(metalID.text.ITEM_CANNOT_BE_OBTAINED, xi.item.CCB_POLYMER) -- These checks happen BEFORE the quest will attempt to play the cs
+                    elseif npcUtil.tradeHasExactly(trade, { xi.item.CARMINE_CHIP, xi.item.CYAN_CHIP, xi.item.GRAY_CHIP }) then
+                        return quest:progressEvent(883)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [883] = function(player, csid, option, npc)
+                    if quest:complete(player) then
+                        player:confirmTrade()
+                    end
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_COMPLETED
+        end,
+
+        [xi.zone.METALWORKS] =
+        {
+            ['Cid'] =
+            {
+                onTrade = function(player, npc, trade)
+                    if
+                        player:hasItem(xi.item.CCB_POLYMER) and
+                        npcUtil.tradeHasExactly(trade, { xi.item.CARMINE_CHIP, xi.item.CYAN_CHIP, xi.item.GRAY_CHIP })
+                    then
+                        player:messageSpecial(metalID.text.ITEM_CANNOT_BE_OBTAINED, xi.item.CCB_POLYMER)
+                    elseif
+                        player:getFreeSlotsCount() == 0 and
```

Comment:
this line could be combined to top logic as return is same, also returns are missing
```lua
if
    (player:hasItem(xi.item.CCB_POLYMER) or player:getFreeSlotsCount() == 0) and
    npcUtil.tradeHasExactly(trade, { xi.item.CARMINE_CHIP, xi.item.CYAN_CHIP, xi.item.GRAY_CHIP })
then
    return quest:messageSpecial(metalID.text.ITEM_CANNOT_BE_OBTAINED, xi.item.CCB_POLYMER)
```

Reply:
same idea for accepted code block